### PR TITLE
Add managed pipeline variables to pipelinegenerator

### PR DESCRIPTION
**Pull request by hallipr**
Add support for managed variables in pipeline generator:

```
meta.platform: java
meta.category: ci
meta.component: storage
meta.autoGenerated: true
```

### CODE REVIEW

File: eng/common/pipelines/templates/steps/prepare-pipelines.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ steps:
         --agentpool Hosted
         --branch refs/heads/$(DefaultBranch)
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/prepare-pipelines.yml:36]: You are going to also need to update https://github.com/Azure/azure-sdk-tools/blob/main/eng/pipelines/pipeline-generation.yml

File: tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs [hunk 2]
```diff
@@ -68,12 +75,23 @@ private int[] ParseIntArray(string[] strs)
...
+        private async Task<VssConnection> GetConnectionAsync()
         {
             if (cachedConnection == null)
             {
-                var pat = Environment.GetEnvironmentVariable(patvar);
-                var credentials = new VssBasicCredential("nobody", pat);
+                VssCredentials credentials;
+                if (string.IsNullOrWhiteSpace(patvar))
+                {
+                    var azureTokenProvider = new AzureServiceTokenProvider();
```
Comment (weshaggard) [inline tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs:85]: Is this using the login information from the local user?

File: eng/common/pipelines/templates/steps/prepare-pipelines.yml (hunk 1 above)
Comment (weshaggard) [reply eng/common/pipelines/templates/steps/prepare-pipelines.yml:36]: You will also need to update https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/install-pipeline-generation.yml so you might need to do these yml files after this PR is merged.

File: tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs (hunk 2 above)
Comment (hallipr) [reply tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs:85]: it is.   It's MSI login, so it'll use the token endpoint or az cli to get the token if available

Comment (hallipr) [reply tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/PipelineGenerationContext.cs:85]: I was unable to generate a PAT with the scopes required.   When querying for the github repo via ADO REST, I received a 401 no matter the scopes in the token.  It allows for PATless development, then PAT or managed identity for prod.

**Comment by hallipr** [comment]
resubmitting from fork

**Pull request closed without merging**